modules/metastorage/src/test/java/org/apache/ignite/internal/metastorage/server/AbstractCompactionKeyValueStorageTest.java

```diff
@@ -17,186 +17,150 @@
-
-        Entry entry1 = storage.get(key, lastRevision - 3);
-        assertTrue(entry1.empty());
+        assertEquals(List.of(3, 5), collectRevisions(FOO_KEY));
+        assertEquals(List.of(5), collectRevisions(BAR_KEY));
+        assertEquals(List.of(4, 6), collectRevisions(SOME_KEY));
     }
 
     @Test
-    public void testCompactEmptyStorage() {
-        storage.compact(clock.now());
+    void testCompactRevision3() {
+        storage.compact(3);
+
+        assertEquals(List.of(5), collectRevisions(FOO_KEY));
+        assertEquals(List.of(5), collectRevisions(BAR_KEY));
+        assertEquals(List.of(4, 6), collectRevisions(SOME_KEY));
     }
 
     @Test
-    public void testCompactionBetweenRevisionsOfOneKey() {
-        byte[] key = key(0);
-        byte[] value11 = keyValue(0, 0);
-        byte[] value12 = keyValue(0, 1);
-
-        storage.put(key, value11, clock.now());
-
-        byte[] key2 = key(1);
-        byte[] value2 = keyValue(1, 0);
-        storage.put(key2, value2, clock.now());
+    void testCompactRevision4() {
+        storage.compact(4);
 
-        HybridTimestamp compactTs = clock.now();
+        assertEquals(List.of(5), collectRevisions(FOO_KEY));
+        assertEquals(List.of(5), collectRevisions(BAR_KEY));
+        assertEquals(List.of(6), collectRevisions(SOME_KEY));
+    }
 
-        storage.put(key, value12, clock.now());
+    @Test
+    void testCompactRevision5() {
+        storage.compact(5);
 
-        storage.compact(compactTs);
+        assertEquals(List.of(5), collectRevisions(FOO_KEY));
+        assertEquals(List.of(), collectRevisions(BAR_KEY));
+        assertEquals(List.of(6), collectRevisions(SOME_KEY));
+    }
 
-        // Both keys should exist, as low watermark's revision is higher than entry11's, but lesser than entry12's,
-        // this means that entry1 is still needed.
-        Entry entry12 = storage.get(key, storage.revision());
-        assertArrayEquals(value12, entry12.value());
+    @Test
+    void testCompactRevision6() {
+        storage.compact(6);
 
-        Entry entry11 = storage.get(key, storage.revision() - 1);
-        assertArrayEquals(value11, entry11.value());
+        assertEquals(List.of(5), collectRevisions(FOO_KEY));
+        assertEquals(List.of(), collectRevisions(BAR_KEY));
+        assertEquals(List.of(), collectRevisions(SOME_KEY));
+    }
 
-        Entry entry2 = storage.get(key2, storage.revision());
-        assertArrayEquals(value2, entry2.value());
+    @Test
+    void testCompactRevisionSequentially() {
+        testCompactRevision1();
+        testCompactRevision2();
+        testCompactRevision3();
+        testCompactRevision4();
+        testCompactRevision5();
+        testCompactRevision6();
     }
 
     @Test
-    public void testInvokeCompactionBeforeAnyEntry() {
-        byte[] key = key(0);
-        byte[] value1 = keyValue(0, 0);
-        byte[] value2 = keyValue(0, 1);
+    public void testCompactEmptyStorage() {
```

Comment:
Why is it about an empty storage as the storage seems to contain something (namely the data added in the 'before each' method)

Reply:
Damn it, you're right, I'll fix it!